modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/imageio/MaskOverviewProvider.java

```diff
@@ -97,62 +97,74 @@
     private ImageInputStreamSpi maskOvrStreamSpi;
 
     private ImageReaderSpi maskOvrReaderSpi;
-
+    
     public MaskOverviewProvider(DatasetLayout layout, File inputFile) throws IOException {
         this(layout, inputFile, (ImageReaderSpi) null);
     }
 
     public MaskOverviewProvider(DatasetLayout layout, File inputFile, ImageReaderSpi suggestedSPI)
             throws IOException {
-        this (layout, inputFile, new SpiHelper(inputFile, suggestedSPI));
+        this (layout, inputFile, new SpiHelper(inputFile.toURI().toURL(), suggestedSPI));
```

Comment:
All toURI().toURL() need to be modified to use URLs.fileToUrl(file) (marked this one, there others, please fix them all)